src/core/net/udp6.cpp

```diff
@@ -353,27 +352,16 @@ otError Udp::SendDatagram(Message &aMessage, MessageInfo &aMessageInfo, uint8_t
 
 otError Udp::HandleMessage(Message &aMessage, MessageInfo &aMessageInfo)
 {
-    otError  error = OT_ERROR_NONE;
-    Header   udpHeader;
-    uint16_t payloadLength;
-    uint16_t checksum;
-
-    payloadLength = aMessage.GetLength() - aMessage.GetOffset();
-
-    // check length
-    VerifyOrExit(payloadLength >= sizeof(Header), error = OT_ERROR_PARSE);
```

Comment:
Adding a note about `diff` here:
The new code performs the reading of `udpHeader` from message first before checksum check. 
With this we can then remove the check `VerifyOrExit(payloadLength >= sizeof(Header)` (it would be covered by header read).